test/e2e/issues_test.go

```diff
@@ -339,6 +339,58 @@ spec:
 		MustApplyYAML(eetestUpdated)
 		FieldShouldContainWithTimeout("eetests.e2e.hnc.x-k8s.io", nsChild, "eetest-sample", ".spec.foo", "bar", 30)
 	})
+
+	It("Should allow objects to be propagated only when a selector is set in Allow mode - issue #16", func() {
+		// Create a simple structure and get an object propagated
+		CreateNamespace(nsParent)
+		CreateNamespace(nsChild)
+		MustRun("kubectl hns set", nsChild, "--parent", nsParent)
+
+		MustRun("kubectl -n", nsParent, "create secret generic my-creds --from-literal=password=iamteamb")
+		// wait 2 seconds to give time for secret to propogate if it was to
+		time.Sleep(2 * time.Second)
```

Comment:
These sleeps _shouldn't_ be necessary - are they? `RunShouldNotContain` should repeatedly retry until it passes (up to a maximum of 2s).